internal/provider/resource_firewall_rule.go

```diff
@@ -169,6 +169,51 @@ func resourceFirewallRule() *schema.Resource {
 				Optional:     true,
 				ValidateFunc: validation.StringInSlice([]string{"match-ipsec", "match-none"}, false),
 			},
+			"weekdays": {
+				Description: "Days of the week the rule is enforced.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"weekdays_negate": {
+				Description: "Negate (inverse) the days of the week the rule is enforced.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+			},
+			"monthdays": {
+				Description: "Days of the month the rule is enforced.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"monthdays_negate": {
+				Description: "Negate (inverse) the days of the month the rule is enforced.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+			},
+			"startdate": {
+				Description: "Date that the rule starts being enforced.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"starttime": {
+				Description: "Time that the rule starts being enforced.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"stopdate": {
+				Description: "Date that the rule stops being enforced.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"stoptime": {
+				Description: "Time that the rule stops being enforced.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
```

Comment:
Can you use a date without a time? Or can you only use them together?

Typically for date/time we use RFC 3339 format, since its a bit easier for international usage. I think if they have to be used together, I'd prefer a single `start` attribute that did both date and time in RFC 3339 format. We have some validation stuff in the SDK for this: https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/helper/validation#IsRFC3339Time

Reply:
If they are used separately (which I believe is the case), we should still probably use date/time formats that are closer to RFC 3339 as that is how Terraform's built-in functions work, for example: https://www.terraform.io/docs/language/functions/timeadd.html